### ISSUE TITLE
duplicate-value false-positive

### ISSUE THREAD

**Issue by starflows** [PyCQA/pylint#8302]
### Bug description

    # pylint: disable=missing-module-docstring
    x = {1, True}
    y = {True, 1}

### Command used

```shell
pylint duplicate_value.py
```

### Pylint output

```shell
************* Module duplicate_value
duplicate_value.py:2:4: W0130: Duplicate value True in set (duplicate-value)
duplicate_value.py:3:4: W0130: Duplicate value 1 in set (duplicate-value)
```

### Expected behavior

no W0130 for `1` and `True`

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.10.9 (main, Dec 19 2022, 17:35:49) [GCC 12.2.0]
```

**Comment by starflows** [comment]
A similar false-positive occurs with

    z = {0, False}

**Comment by mbyrnepr2** [comment]
Thanks @starflows. This looks expected when I look at the Python behaviour:
```python
>>> {True, True, 1, 1}
{True}
>>> {1, 1, True, True}
{1}

```

**Comment by starflows** [comment]
Thanks @mbyrnepr2 . I was not aware of that. In that case the pylint warning seems correct. Also my usecase works since

    >>> True in {1}
    True
